cfme/tests/automate/custom_button/test_buttons.py

```diff
@@ -351,7 +366,44 @@ def test_custom_button_simulation():
     Bugzilla:
         1535215
     """
-    pass
+    if button_tag == "Evm":
+        btn_type = "User"
+        obj = appliance.collections.users.instantiate(name="Administrator")
+    else:
+        btn_type = "Provider"
+        obj = provider
+
+    gp = appliance.collections.button_groups.instantiate(
+        text="[Unassigned Buttons]", hover="Unassigned buttons", type=btn_type
+    )
+
+    button = gp.buttons.create(
+        text="Btn_{}".format(fauxfactory.gen_alphanumeric(2)),
+        hover="Hover_{}".format(fauxfactory.gen_alphanumeric(2)),
+        system="Request",
+        request="InspectMe",
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    # Clear the automation log
+    assert appliance.ssh_client.run_command('echo -n "" > /var/www/miq/vmdb/log/automation.log')
```

Comment:
:+1: :100:

Reply:
This is ultimately a bit of a hack - there are tools for dealing with the logs, and grepping for certain content. It automatically handles only checking the log during the test function execution. Okay for now as `log_request_check` already exists, but it should be replaced.